### DISCUSSION TITLE
Differences between stm32u5 based ST Nucleo U575ZI Q and b_u585i_iot02a Boards

### DISCUSSION THREAD

**Discussion by zafersn** [zephyrproject-rtos/zephyr#64276] [Q&A]
Hi There, 

1. Regardless of the dev board HW design. I am wondering Why doesn't the 575zi board have board options/configurations such as tfm build configs and memory partitions while 585 does.  I think this is a common feature of this chip family.

2.  This is from the b_u585i_iot02a_ns.dts file. I just wanted to add all allocated memory regions and here is the result. ~16kb is missed while the sum of all regions' size is exactly 1mb in the b_u585i_iot02a.dts. Curious why the non-secure config is missing 16kb (normally 2mb = 1024\*1024\*2 byte = 2,097,152 )
![image](https://github.com/zephyrproject-rtos/zephyr/assets/13236364/a2932915-b043-478d-989a-5ebaf242fbb8)

3.  Why does the b_u585i_iot02a_ns have a 224kb boot portion size? Is it configurable for the board option _ns 
4. Can I use the entire 2mb flash configuration for the u575zi board?  Do I really need to have _ns configuration for the u575zi board?

Thank you so much for your help.
Kind regards

**Comment by erwango** [comment]
> 1. I am wondering Why doesn't the 575zi board have board options/configurations such as tfm build configs and memory partitions while 585 does. I think this is a common feature of this chip family.

To enable TF-M, you need that the board is enabled in TF-M, which is not the case is upstream TF-m for nucleo U5 board: https://github.com/zephyrproject-rtos/trusted-firmware-m/tree/master/platform/ext/target/stm

> 2. 3. 4.

I didn't get in the details of your questions, but Zephyr partitioning should be aligned with the partitioning defined in TFM in https://github.com/zephyrproject-rtos/trusted-firmware-m/blob/master/platform/ext/target/stm/b_u585i_iot02a/include/flash_layout.h
(Note that the summary section in comment is wrong and you should rather check the symbols definitions).

**Reply by zafersn** [reply]
> To enable TF-M, you need that the board is enabled in TF-M, which is not the case is upstream TF-m for nucleo U5 board: https://github.com/zephyrproject-rtos/trusted-firmware-m/tree/master/platform/ext/target/stm

get it. is tf-m support for the nucleo u5 on your agenda? And Can you please elaborate on what are the requirements/jobs need to be done to have tf-m for U5?

> I didn't get in the details of your questions,

Sorry, it is my mistake, let me explain again.

1. I just wanted to calculate the total allocated memory size in the b_u585i_iot02a_ns.dts (none secure)and even though it should be 2mb, it was 16kb missing. Why?
2. Is there any specific reason to allocate 224kb for the boot partition? was it not enough 64kb for the non secure config?
3. Is it possible to use the total size of the flash memory of 2mb on the nucleo u575 without none secure config?
    -  Like basically, will it work if I copy the flash partitions definition/configs(2mb) from the b_u585i_iot02a_ns.dts to nucleo_u575.dts?

Thank you for your help!

**Reply by erwango** [reply]
> get it. is tf-m support for the nucleo u5 on your agenda? And Can you please elaborate on what are the requirements/jobs need to be done to have tf-m for U5?

Not on my side. But TF-M is an open source project as well, and I guess that contributions would be welcomed

> Sorry, it is my mistake, let me explain again.

My bad if it wasn't clear, I didn't want to go in the details of your question, but I guess that most of it could be solved by this:
Zephyr partitioning should be aligned with the partitioning defined in TFM in https://github.com/zephyrproject-rtos/trusted-firmware-m/blob/master/platform/ext/target/stm/b_u585i_iot02a/include/flash_layout.h
(Note that the summary section in comment is wrong and you should rather check the symbols definitions).
